sources/osgUtil/LineSegmentIntersectFunctor.js

```diff
@@ -44,272 +40,338 @@ var LineSegmentIntersectFunctor = function() {
+                    sN = 0.0; // force using point P0 on segment S1
+                    sD = 1.0; // to prevent possible division by 0.0 later
+                    tN = e;
+                    tD = c;
+                } else {
+                    // get the closest points on the infinite lines
+                    sN = b * e - c * d;
+                    tN = a * e - b * d;
+                    if (sN < 0.0) {
+                        // sc < 0 => the s=0 edge is visible
+                        sN = 0.0;
+                        tN = e;
+                        tD = c;
+                    } else if (sN > sD) {
+                        // sc > 1  => the s=1 edge is visible
+                        sN = sD;
+                        tN = e + b;
+                        tD = c;
+                    }
+                }
+
+                if (tN < 0.0) {
+                    // tc < 0 => the t=0 edge is visible
+                    tN = 0.0;
+                    // recompute sc for this edge
+                    if (-d < 0.0) sN = 0.0;
+                    else if (-d > a) sN = sD;
+                    else {
+                        sN = -d;
+                        sD = a;
+                    }
+                } else if (tN > tD) {
+                    // tc > 1  => the t=1 edge is visible
+                    tN = tD;
+                    // recompute sc for this edge
+                    if (-d + b < 0.0) sN = 0;
+                    else if (-d + b > a) sN = sD;
+                    else {
+                        sN = -d + b;
+                        sD = a;
+                    }
+                }
+                // finally do the division to get sc and tc
+                var sc = Math.abs(sN) < epsilon ? 0.0 : sN / sD;
+                var tc = Math.abs(tN) < epsilon ? 0.0 : tN / tD;
+
+                // get the difference of the two closest points
+                vec3.scaleAndAdd(closest0, v0, u, sc);
+                vec3.scaleAndAdd(closest1, this._start, v, tc);
+
+                var sqrDistance = vec3.sqrDist(closest0, closest1);
+                if (sqrDistance > this._threshold * this._threshold) {
+                    return;
+                }
+
+                var intersection = this._createIntersection();
+                intersection._i1 = p0;
+                intersection._i2 = p1;
+                intersection._r1 = 1.0 - tc;
+                intersection._r2 = tc;
+
+                vec3.copy(intersection._localIntersectionPoint, closest1);
+                vec3.copy(intersection._localIntersectionNormal, vec3.ONE);
+                intersection._ratio = tc;
+            };
+        })(),
 
-            if (e[2] > zmax) {
-                // clip e to zMax.
-                vec3.scaleAndAdd(e, s, invZ, zmax - s[2]);
-            }
-        } else {
-            if (s[2] < zmin) return false;
-            if (e[2] > zmax) return false;
+        intersectTriangle: (function() {
```

Comment:
It seems threshold can't be used with triangles, it's a bit confusing, From a user perspective I would expect this to work as well. 

Maybe adding comments to the header could clarify a bit.

Reply:
yep